channeldb/invoices.go

```diff
@@ -341,7 +343,9 @@ func (d *DB) LookupInvoice(_ context.Context, ref invpkg.InvoiceRef) (
 
 		// An invoice was found, retrieve the remainder of the invoice
 		// body.
-		i, err := fetchInvoice(invoiceNum, invoices, setID)
+		i, err := fetchInvoice(
+			invoiceNum, invoices, []*invpkg.SetID{setID}, true,
```

Comment:
On one hand, this could be considered a breaking change, potentially unsuitable for a minor release. On the other hand, IMO it was an existing bug as `LookupInvoice` when a `setID` is set should only return the items relevant to that `setID` sub-invoice.

Reply:
I think this bug was more of a side effect of how the invoice update was saved back to KV db, since there we serialize the whole thing into one blob. I'd maybe therefore categorize it as a fix and it is also better to only return the relevant AMP state.